BlueDBOnDisk/src/main/java/io/bluedb/disk/collection/index/BlueIndexOnDisk.java

```diff
@@ -112,10 +124,37 @@ public void rollup(Range range) throws BlueDbException {
 		return segmentManager;
 	}
 
-	private List<IndexCompositeKey<I>> toCompositeKeys(BlueKey destination, T newItem) throws BlueDbException {
+	private List<IndividualChange<BlueKey>> toIndexChanges(Collection<IndividualChange<T>> changes) {
+		return changes.stream()
+				.map( (IndividualChange<T> change) -> toIndexChanges(change) )
+				.flatMap(List::stream).collect(Collectors.toList());
+	}
+
+	private List<IndividualChange<BlueKey>> toIndexChanges(IndividualChange<T> change) {
+		List<IndexCompositeKey<I>> compositeKeys = toCompositeKeys(change);
+		BlueKey underlyingKey = change.getKey();
+		return toIndexChanges(compositeKeys, underlyingKey);
+	}
+
+	private static <I extends ValueKey> List<IndividualChange<BlueKey>> toIndexChanges(List<IndexCompositeKey<I>> compositeKeys, BlueKey destinationKey) {
+		List<IndividualChange<BlueKey>> indexChanges = new ArrayList<>();
+		for (IndexCompositeKey<I> compositeKey: compositeKeys) {
+			IndividualChange<BlueKey> indexChange = IndividualChange.insert(compositeKey, destinationKey);
+			indexChanges.add(indexChange);
+		}
+		return indexChanges;
+	}
+
+	private List<IndexCompositeKey<I>> toCompositeKeys(IndividualChange<T> change) {
+		BlueKey destinationKey = change.getKey();
+		T newValue = change.getNewValue();
+		return toCompositeKeys(destinationKey, newValue);
+	}
+
+	private List<IndexCompositeKey<I>> toCompositeKeys(BlueKey destination, T newItem) {
 		List<I> indexKeys = keyExtractor.extractKeys(newItem);
 		CheckedFunction<I, IndexCompositeKey<I>> indexToComposite = (indexKey) -> new IndexCompositeKey<I>(indexKey, destination);
-		List<IndexCompositeKey<I>> compositeKeys = Blutils.map(indexKeys, indexToComposite);
+		List<IndexCompositeKey<I>> compositeKeys = Blutils.mapIgnoringExceptions(indexKeys, indexToComposite);

```

Comment:
Why are you no longer throwing BlueDbException? How does that affect the places that were calling that before? (Add and Remove)

Reply:
Blutils.map was created to handle mappers that can throw exceptions and Blutils.mapIgnoringExceptions was created to handle those and ignore the exceptions.  I used it because I needed a mapper that didn't throw exceptions and Java streams is so clunky.  But I think mapIgnoringExceptions  is not appropriate.

I replaced a couple of these calls with a helper function and another one with a stream